pkg/networkservice/common/clienturl/client.go

```diff
@@ -75,11 +75,16 @@ func (u *clientURLClient) init() error {
 			u.dialErr = errors.New("cannot dial nil clienturl.ClientURL(ctx)")
 			return
 		}
+
+		ctx, cancel := context.WithTimeout(u.ctx, dialTimeout)
```

Comment:
Why are we introducing a new arbitrary dialTimeout instead of deriving it from the ctx of the request?

Reply:
It will break discover + roundrobin, because it will wait for the first dial until requests lives and so there won't be any chance to try someone else.